app/src/main/java/net/gsantner/markor/format/plaintext/PlaintextTextActions.java

```diff
@@ -10,56 +10,62 @@
 
 import android.app.Activity;
+import android.support.annotation.StringRes;
 import android.view.HapticFeedbackConstants;
 import android.view.View;
-import android.view.ViewGroup;
 
 import net.gsantner.markor.R;
+import net.gsantner.markor.activity.ActionItem;
 import net.gsantner.markor.format.general.CommonTextActions;
 import net.gsantner.markor.model.Document;
 import net.gsantner.markor.ui.hleditor.TextActions;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class PlaintextTextActions extends TextActions {
 
     public PlaintextTextActions(Activity activity, Document document) {
         super(activity, document);
     }
 
+    @Override
+    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
+        return runCommonTextAction(action);
+    }
 
     @Override
-    public void appendTextActionsToBar(ViewGroup barLayout) {
-        if (barLayout.getChildCount() == 0) {
-            setBarVisible(barLayout, true);
+    protected ActionCallback getActionCallback(@StringRes int keyId) {
+        return new PlaintextTextActionImpl(keyId);
+    }
 
-            // Regular actions
-            for (int[] actions : TMA_ACTIONS) {
-                PlaintextTextActionImpl actionCallback = new PlaintextTextActionImpl(actions[0]);
-                appendTextActionToBar(barLayout, actions[1], actions[2], actionCallback, actionCallback);
-            }
-        }
+    @Override
+    public List<ActionItem> getActiveActionList() {
+        return Arrays.asList(TMA_ACTIONS);
     }
 
     @Override
-    public boolean runAction(String action, boolean modLongClick, String anotherArg) {
-        return runCommonTextAction(action);
+    protected @StringRes int getFormatActionsKey() {
+        return R.string.pref_key__plaintext__action_keys;
     }
 
     //
     //
     //
-    private static final int[][] TMA_ACTIONS = {
-            {R.string.tmaid_common_checkbox_list, R.drawable.ic_check_box_black_24dp, R.string.check_list},
-            {R.string.tmaid_common_unordered_list_char, R.drawable.ic_list_black_24dp, R.string.unordered_list},
-            {R.string.tmaid_common_ordered_list_number, R.drawable.ic_format_list_numbered_black_24dp, R.string.ordered_list},
-            {R.string.tmaid_common_jump_to_bottom, CommonTextActions.ACTION_JUMP_BOTTOM_TOP_ICON, R.string.jump_to_bottom},
-            {R.string.tmaid_common_delete_lines, CommonTextActions.ACTION_DELETE_LINES_ICON, R.string.delete_lines},
-            {R.string.tmaid_common_open_link_browser, CommonTextActions.ACTION_OPEN_LINK_BROWSER__ICON, R.string.open_link},
-            {R.string.tmaid_common_attach_something, R.drawable.ic_attach_file_black_24dp, R.string.attach},
-            {R.string.tmaid_common_special_key, CommonTextActions.ACTION_SPECIAL_KEY__ICON, R.string.special_key},
-            {R.string.tmaid_common_time, R.drawable.ic_access_time_black_24dp, R.string.date_and_time},
+
+    public static final ActionItem[] TMA_ACTIONS = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5344500)